topi/python/topi/cuda/sort.py

```diff
@@ -237,6 +237,53 @@ def sort_nms_ir(data, valid_count, output, axis, is_ascend):
 
     return ib.get()
 
+def argsort_nms_thrust(data, valid_count, axis=-1, is_ascend=1, dtype="float32"):
+    """Performs sorting along the given axis and returns an array of indicies
+    having same shape as an input array that index data in sorted order.
+
+    Parameters
+    ----------
+    data: tvm.te.Tensor
+        The input array.
+
+    valid_count : tvm.te.Tensor, optional
+        The number of valid elements to be sorted.
+
+    axis : int, optional
+        Axis long which to sort the input tensor.
+
+    is_ascend : boolean, optional
+        Whether to sort in ascending or descending order.
+
+    dtype : string, optional
+        DType of the output indices.
+
+    Returns
+    -------
+    out : tvm.te.Tensor
+        The output of this function.
+    """
+    if axis < 0:
+        axis = len(data.shape) + axis
```

Comment:
Add check for the axis.  For example,
```
assert axis==len(data.shape)-1, "Supports sorting along the last axis only"
```

Reply:
We don't only support sorting along the last axis.

Reply:
Shall I add `assert 0 <= axis < ndim` after this line?

Reply:
In original sort_nms, we do support `axis` other than -1, I'll add your swap operations in this operator.